### PR TITLE
resource id fix

### LINKED ISSUE #2501: Invalid URL from cubemap resource handler
Example scene: https://playcanvas.com/editor/scene/480301.

Open the Cubemap folder in the Asset Panel and notice all the errors about a 'scene not found'. The issue is that each asset has its ID as the URL so the browser tries to load it as a relative URL.

This is set in `engine/src/resources/cubemap.js` line 321.

### PR DESCRIPTION
**Pull request by slimbuck**
Fixes #2501

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/resources/cubemap.js [hunk 1]
```diff
@@ -297,7 +297,7 @@ Object.assign(CubemapHandler.prototype, {
             } else if (self.compareAssetIds(assetId, loadedAssetIds[i])) {
                 // asset id hasn't changed from what is currently set
                 onReady(i, loadedAssets[i]);
-            } else if (parseInt(assetId, 10) === assetId) {
+            } else if (parseInt(assetId, 10) == assetId) {
```
Comment (willeastcott) [inline src/resources/cubemap.js:300]: Why do === to ==?

Comment (willeastcott) [inline src/resources/cubemap.js:300]: This is a bit scary. I really dislike double equals and so does ESlint it seems:

https://eslint.org/docs/rules/eqeqeq

Some coder in the future might just decide to 'improve' the code and switch to triple equals, not understanding the rationale. Is there not a clearer way to deal with this?

Comment (slimbuck) [reply src/resources/cubemap.js:300]: I can add a comment with explanation?

Comment (slimbuck) [reply src/resources/cubemap.js:300]: something like:
```javascript
} else if (parseInt(assetId, 10) == assetId) {  // NOTE: use '==' to handle assetId being 123 or '123'.
```

Comment (slimbuck) [reply src/resources/cubemap.js]: wrapped the test in a function and used explicit '===' instead

**Review by willeastcott — APPROVED** [review]

**Pull request merged**